client-misk/src/main/kotlin/app/cash/backfila/client/misk/BackfilaModule.kt

```diff
@@ -4,37 +4,44 @@ import app.cash.backfila.client.misk.client.BackfilaClientConfig
 import app.cash.backfila.client.misk.internal.BackfilaClient
 import app.cash.backfila.client.misk.internal.BackfilaStartupConfigurator
 import app.cash.backfila.client.misk.internal.RealBackfilaClient
+import com.google.inject.Binder
 import com.google.inject.BindingAnnotation
 import com.google.inject.Provides
 import com.google.inject.TypeLiteral
+import com.google.inject.multibindings.MapBinder
 import com.squareup.moshi.Moshi
 import com.squareup.moshi.kotlin.reflect.KotlinJsonAdapterFactory
+import misk.ServiceModule
+import misk.inject.KAbstractModule
 import javax.inject.Singleton
 import kotlin.reflect.KClass
 import kotlin.reflect.jvm.jvmName
-import misk.ServiceModule
-import misk.inject.KAbstractModule
 
 /**
  * Backfila-using applications install this and either [EmbeddedBackfilaModule] (testing and
- * development) or [RealBackfilaModule] (staging and production).
+ * development) or [BackfilaClientModule] (staging and production).
  */
 class BackfilaModule(
-  private val config: BackfilaClientConfig,
-  private val backfills: List<KClass<out Backfill<*, *>>>
+    private val config: BackfilaClientConfig,
+    @Deprecated(message = "Multibind backfills instead")
+    private val backfills: List<KClass<out Backfill<*, *>>>? = null
 ) : KAbstractModule() {
   override fun configure() {
     bind<BackfilaClientConfig>().toInstance(config)
 
     bind<BackfilaClient>().to<RealBackfilaClient>()
 
-    val map = mutableMapOf<String, KClass<out Backfill<*, *>>>()
-    for (backfill in backfills) {
-      map[backfill.jvmName] = backfill
+    if (backfills == null) {
+      mapBinder(binder())
+    } else {
+      val map = mutableMapOf<String, KClass<out Backfill<*, *>>>()
+      for (backfill in backfills) {
+        map[backfill.jvmName] = backfill
+      }
+      bind(object : TypeLiteral<MutableMap<String, KClass<out Backfill<*, *>>>>() {})
```

Comment:
why not multibind them here? Then you can gradually transition from non-multibindings to multibindings

Reply:
doh, fixed